src/daemon/org.projectatomic.rpmostree.xml

```diff
@@ -0,0 +1,195 @@
+<!DOCTYPE node PUBLIC
+"-//freedesktop//DTD D-BUS Object Introspection 1.0//EN"
+"http://www.freedesktop.org/standards/dbus/1.0/introspect.dtd">
+<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
+
+  <interface name="org.projectatomic.rpmostree.Manager">
+    <property name="DefaultSysroot" type="o" access="read"/>
+
+    <!--
+         Returns a object path for the Sysroot
+         at the give path. If the path is not
+         not a valid sysroot an error is returned.
+    -->
+    <method name="GetSysroot">
+      <arg name="path" type="s" direction="in"/>
+
+      <arg name="sysroot" type="o" direction="out"/>
+    </method>
+  </interface>
+
+  <interface name="org.projectatomic.rpmostree.Sysroot">
+
+    <!--
+      The basic interface for working with an OSTree.
+
+      * Only one OSTree related action that mutates data can run at once.
+      * All DBus methods that mutate data return immediately, and fire
+        signals when complete.
+      * The signals are never fired before the method returns.
+      * The daemon never exits while a long running task is running.
+    -->
+
+    <property name="SysrootPath" type="s" access="read"/>
+
+    <!--
+         Current Deployment
+    -->
+    <property name="DefaultDeployment" type="o" access="read"/>
+
+    <!--
+         Deployment that the system was booted with.
+    -->
+    <property name="BootedDeployment" type="o" access="read"/>
+
+    <!--
+         Type of update in progress. Possible values are:
+           * rpm-pull
+           * upgrade
+           * rebase
+           * rollback
+    -->
+    <property name="UpdateRunning" type="s" access="read"/>
+
+    <!--
+         Cancels a running update.
+    -->
+    <method name="CancelUpdate">
+    </method>
+
+    <!--
+         Returns a list deployment object paths sorted by
+         their index. So that the first one is always
+         the default for the given os.
+    -->
+    <method name="GetDeployments">
+      <arg name="options" type="a{sv}" direction="in"/>
+      <arg name="deployments" type="ao" direction="out"/>
+    </method>
+
+    <!--
+         Returns a list of current RefSpecs and
+         their object paths.
+    -->
+    <method name="GetRefSpecs">
+      <arg name="refspecs" type="a{so}" direction="out"/>
+    </method>
+
+    <!--
+         Add a new refspec
+
+         Doesn't pull actual data just ensures
+         that the given refspec exists and
+         adds it to list of known refspecs
+    -->
+
+    <method name="AddRefSpec">
+      <arg name="options" type="a{sv}" direction="in"/>
+      <arg name="refspec" type="s" direction="in"/>
+
+      <!--
+           The refspec we created.
+      -->
+      <arg name="refspec" type="o" direction="out"/>
```

Comment:
Suggest naming this something different from the input arg -- `refspec_path` maybe?